tests/sentry/incidents/test_tasks.py

```diff
@@ -208,7 +209,8 @@ def test_missing_project(self):
         )
 
     def test(self):
-        with patch.object(AlertRuleTriggerAction, "_factory_registrations", new={}):
+        suspended_registrations = TemporaryAlertRuleTriggerActionRegistry.suspend()
```

Comment:
what is the difference between patching, and temporarily overriding the registrations in tests?

Reply:
They're functionally equivalent. My gripe with the old way wasn't about `patch.object`, but that the thing we were patching was underscore-prefixed and should have been private. IMHO, the privacy convention applies to unit tests too, and we were breaking it here.

My approach with `TemporaryAlertRuleTriggerActionRegistry` is equally kludgey, but my goal was to isolate it into one helper class that lives in a `testutils` package, so as to more clearly communicate "this is some special test-only magic".

As for the try-finally structure versus the `with` block, I actually like the `with` better, but the `suspend` and `restore` methods were necessary because of the other unit tests that want to call them during `setUp` and `tearDown`. If we value the `with` structure, I could put add a helper method with `@contextmanager` that lets us continue to use it.

Reply:
https://github.com/getsentry/sentry/pull/74936/commits/0e792376bfd8e74b3f2aad40a964d00f4dde24ca